clients/drcachesim/tools/record_view.cpp

```diff
@@ -0,0 +1,215 @@
+    } else if (trace_type == TRACE_TYPE_PID) {
+        uint pid = (uint)entry.addr;
+        std::cerr << trace_type_name << ", pid: " << pid << "\n";
+    } else if (trace_type == TRACE_TYPE_MARKER) {
+        /* XXX i#6751: we have a lot of different types of markers; we should use some
+         * kind of dispatching mechanism to print a more informative output for each of
+         * them.  For now we do so only for a few markers here.
+         */
+        trace_marker_type_t trace_marker_type = (trace_marker_type_t)entry.size;
+        std::string trace_marker_name =
+            std::string(trace_marker_names[trace_marker_type]);
+        addr_t trace_marker_value = entry.addr;
+        if (trace_marker_type == TRACE_MARKER_TYPE_FILETYPE) {
+            std::string file_type =
+                std::string(trace_arch_string((offline_file_type_t)trace_marker_value));
+            std::cerr << trace_type_name << ", trace_marker_type: " << trace_marker_name
+                      << ", trace_marker_value: " << file_type << "\n";
+        } else if (trace_marker_type == TRACE_MARKER_TYPE_VERSION) {
+            trace_version_t trace_version = (trace_version_t)entry.addr;
+            std::string trace_version_name =
+                std::string(trace_version_names[trace_version]);
+            std::cerr << trace_type_name << ", trace_marker_type: " << trace_marker_name
+                      << ", trace_marker_value: " << trace_version_name << "\n";
+        } else { // For all remaining markers we print their value in hex.
+            std::cerr << trace_type_name << ", trace_marker_type: " << trace_marker_name
+                      << ", trace_marker_value: 0x" << std::hex << trace_marker_value
+                      << std::dec << "\n";
+        }
+    } else if (trace_type == TRACE_TYPE_ENCODING) {
+        unsigned short num_encoding_bytes = entry.size;
+        std::cerr << trace_type_name << ", num_encoding_bytes: " << num_encoding_bytes
+                  << ", encoding_bytes: 0x" << std::hex;
+        /* Print encoding byte by byte (little-endian).
+         */
+        for (int i = num_encoding_bytes - 1; i >= 0; --i) {
+            uint encoding_byte = static_cast<uint>(entry.encoding[i]);
+            std::cerr << encoding_byte;
+        }
+        std::cerr << std::dec << "\n";
+    } else if (trace_type == TRACE_TYPE_INSTR_BUNDLE) {
+        unsigned short num_instructions_in_bundle = entry.size;
+        std::cerr << trace_type_name
+                  << ", num_instructions_in_bundle: " << num_instructions_in_bundle
+                  << ", instrs_length:";
+        /* Print length of each instr in the bundle.
+         */
+        for (int i = 0; i < num_instructions_in_bundle; ++i) {
+            unsigned char instr_length = entry.length[i];
+            std::cerr << " " << instr_length;
+        }
+        std::cerr << "\n";
+    } else if (type_is_instr(trace_type)) {
```

Comment:
clients/drcachesim/tools/view.cpp disambiguates various kinds of instrs. We should share that logic with this tool.

Reply:
Mhm, it's actually just renaming trace_type_names[], which is weird, because every other tool uses trace_type_names[], but view.cpp does not.
Should we switch this around and have view.cpp use trace_type_names[]?
Or change trace_type_names[] with what view.cpp prints?
Or does this break too much documentation and/or other assumptions I'm not aware of?
(If we decide to go this way, the aforementioned changes should go in a separate PR IMHO).